src/status_im/extensions/ethereum.cljs

```diff
@@ -0,0 +1,37 @@
+(ns status-im.extensions.ethereum
+  (:require [status-im.utils.handlers :as handlers]
+            [re-frame.core :as re-frame]
+            [status-im.models.wallet :as models.wallet]
+            [status-im.utils.ethereum.abi-spec :as abi-spec]))
+
+(handlers/register-handler-fx
+ :extensions/transaction-on-result
+ (fn [{db :db} [_ on-result id result method]]
+   {:dispatch-n (concat
+                 [[:navigate-to-clean :wallet-transaction-sent]]
+                 (when on-result
+                   [(on-result {:error nil :result result})]))}))
```

Comment:
I would much prefer to use:
```clj
(fx/merge cofx
          (when on-result
            {:dispatch (on-result {:error nil :result result}))}
          (navigation/navigate-to-clean :wallet-transaction-sent))
```

Using `:dispatch-n` is generally dangerous, especially when used to encode successions of steps, because it relies on implementation detail of `re-frame` and not interface, interface says there is no particular order in which the events will be dispatched.

Using some unnecessary events is unfortunately always necessary with extensions, because it was decided that event is the only logic building block exposed to the extension writer, but we should still strive to keep it minimal.

Reply:
thanks, i'll fix that, but you could suggest change next time, using new github feature :)

Reply:
Sorry, didn't know about that really, will try it :)